microbench/src/main/java/io/netty/microbench/buffer/BufferOffsetOfBenchmark.java

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright 2021 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package io.netty.microbench.buffer;
+
+import io.netty.buffer.api.Buffer;
+import io.netty.buffer.api.BufferAllocator;
+import io.netty.buffer.api.MemoryManager;
+import io.netty.buffer.api.bytebuffer.ByteBufferMemoryManager;
+import io.netty.buffer.api.unsafe.UnsafeMemoryManager;
+import io.netty.microbench.util.AbstractMicrobenchmark;
+import io.netty.util.internal.SuppressJava6Requirement;
+import org.openjdk.jmh.annotations.Benchmark;
+import org.openjdk.jmh.annotations.Fork;
+import org.openjdk.jmh.annotations.Level;
+import org.openjdk.jmh.annotations.Measurement;
+import org.openjdk.jmh.annotations.OutputTimeUnit;
+import org.openjdk.jmh.annotations.Param;
+import org.openjdk.jmh.annotations.Scope;
+import org.openjdk.jmh.annotations.Setup;
+import org.openjdk.jmh.annotations.State;
+import org.openjdk.jmh.annotations.TearDown;
+import org.openjdk.jmh.annotations.Warmup;
+
+import java.util.SplittableRandom;
+import java.util.concurrent.TimeUnit;
+
+@State(Scope.Benchmark)
+@OutputTimeUnit(TimeUnit.MICROSECONDS)
+@Fork(value = 2, jvmArgsAppend = {
+        "-Dio.netty.tryReflectionSetAccessible=true",
+        "--add-opens", "java.base/java.nio=ALL-UNNAMED",
+        "--add-opens", "java.base/jdk.internal.misc=ALL-UNNAMED",
+        "-XX:+UnlockDiagnosticVMOptions", "-XX:+DebugNonSafepoints" })
+@Warmup(iterations = 5, time = 1)
+@Measurement(iterations = 8, time = 1)
+public class BufferOffsetOfBenchmark extends AbstractMicrobenchmark {
+
+    @Param({
+            "7",
+            "16",
+            "23",
+            "32" ,
+//            "500",
+    })
+    int size;
+
+    @Param({
+            "4",
+            "11",
+    })
+    int logPermutations;
+
+    @Param({ "1" })
+    int seed;
+
+    int permutations;
```

Comment:
private ?

Reply:
Looking around, the parameters are more often public. These are package private because this was copied from `ByteBufIndexOfBenchmark`